src/classes/ADDR_CannotDelete_TEST.cls

```diff
@@ -0,0 +1,452 @@
+        insert contacts;
+
+        List<Address__c> addresses = UTIL_UnitTestData_TEST.getMultipleTestAddresses(6);
+
+        for (Integer index = 0; index < 3; index++) {
+            addresses[index].Parent_Account__c = accounts[0].Id;
+        }
+
+        for (Integer index = 3; index < 6; index++) {
+            addresses[index].Parent_Contact__c = contacts[0].Id;
+        }
+        
+        insert addresses;
+        
+        Test.startTest();
+        Database.DeleteResult[] results = Database.delete(addresses, FALSE);
+        Test.stopTest();
+
+        List<Address__c> returnAddresses = [
+            SELECT Id
+            FROM Address__c
+            WHERE Id IN :addresses
+        ];
+
+        System.assert(returnAddresses.isEmpty());
+
+        for (Database.DeleteResult result : results) {
+            System.assertEquals(TRUE, result.isSuccess());
+        }
+    }
+
+    /*********************************************************************************************************
+    * @description Test method to test that when Prevent_Address_Deletion__c is disabled in Hierarchy Settings, 
+    * and Address does not have a child Account or Contact record associated to it, the Address record can be deleted.
+    */
+    @isTest
+    private static void addressCanDeleteWithoutChildrenPreventDeleteOff() {
+            
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c
+                                                            (Account_Processor__c = ADDR_CannotDelete_TEST.hhAccRecordTypeID,
+                                                            Prevent_Address_Deletion__c = FALSE));
+
+        
+        List<Account> accounts = UTIL_UnitTestData_TEST.getMultipleTestAccounts(1, ADDR_CannotDelete_TEST.hhAccRecordTypeID);
+        insert accounts;
+
+        List<Contact> contacts = UTIL_UnitTestData_TEST.getMultipleTestContacts(1);
+        insert contacts;
+
+        List<Address__c> addresses = UTIL_UnitTestData_TEST.getMultipleTestAddresses(6);
+
+        for (Integer index = 0; index < 3; index++) {
+            addresses[index].Parent_Account__c = accounts[0].Id;
+        }
+
+        for (Integer index = 3; index < 6; index++) {
+            addresses[index].Parent_Contact__c = contacts[0].Id;
+        }
+        
+        insert addresses;
+        
+        Test.startTest();
+        Database.DeleteResult[] results = Database.delete(addresses, FALSE);
+        Test.stopTest();
+
+        List<Address__c> returnAddresses = [
+            SELECT Id
+            FROM Address__c
+            WHERE Id IN :addresses
+        ];
+
+        System.assert(returnAddresses.isEmpty());
+
+        for (Database.DeleteResult result : results) {
+            System.assertEquals(TRUE, result.isSuccess());
+        }
+    }
+
+    /*********************************************************************************************************
```

Comment:
@skosgi1 I added the following three test cases (starting here) to test the hasChildRecords method in ADDR_CannotDelete_TDTM